ms_test.py

```diff
@@ -0,0 +1,346 @@
+            if pModel.process_weight is not None:
+                pModel.process_weight(sym, arg_params, aux_params)
+
+            # merge batch normalization to speedup test
+            from utils.graph_optimize import merge_bn
+
+            sym, arg_params, aux_params = merge_bn(sym, arg_params, aux_params)
+            sym.save(pTest.model.prefix + "_test.json")
+
+            # infer shape
+            worker_data_shape = dict(loader.provide_data + loader.provide_label)
+            for key in worker_data_shape:
+                worker_data_shape[key] = (pKv.batch_image,) + worker_data_shape[key][1:]
+            arg_shape, _, aux_shape = sym.infer_shape(**worker_data_shape)
+            _, out_shape, _ = sym.get_internals().infer_shape(**worker_data_shape)
+            out_shape_dict = list(zip(sym.get_internals().list_outputs(), out_shape))
+            _, out_shape, _ = sym.infer_shape(**worker_data_shape)
+            terminal_out_shape_dict = zip(sym.list_outputs(), out_shape)
+            print('parameter shape')
+            print(pprint.pformat([i for i in out_shape_dict if not i[0].endswith('output')]))
+            print('intermediate output shape')
+            print(pprint.pformat([i for i in out_shape_dict if i[0].endswith('output')]))
+            print('terminal output shape')
+            print(pprint.pformat([i for i in terminal_out_shape_dict]))
+
+            for i in pKv.gpus:
+                ctx = mx.gpu(i)
+                mod = DetModule(sym, data_names=data_names, context=ctx)
+                mod.bind(data_shapes=loader.provide_data, for_training=False)
+                mod.set_params(arg_params, aux_params, allow_extra=False)
+                execs.append(mod)
+
+        all_outputs = []
+
+        if index_split == 0:
+            def eval_worker(exe, data_queue, result_queue):
+                while True:
+                    batch = data_queue.get()
+                    exe.forward(batch, is_train=False)
+                    out = [x.asnumpy() for x in exe.get_outputs()]
+                    result_queue.put(out)
+
+
+            for exe in execs:
+                workers.append(Thread(target=eval_worker, args=(exe, data_queue, result_queue)))
+            for w in workers:
+                w.daemon = True
+                w.start()
+
+        import time
+
+        t1_s = time.time()
+
+
+        def data_enqueue(loader, data_queue):
+            for batch in loader:
+                data_queue.put(batch)
+
+
+        enqueue_worker = Thread(target=data_enqueue, args=(loader, data_queue))
+        enqueue_worker.daemon = True
+        enqueue_worker.start()
+
+        for index in range(loader.total_record):
+            r = result_queue.get()
+
+            rid, id, info, post_cls_score, post_box, post_cls, mask, iou_pred = r
+            rid, id, info, post_cls_score, post_box, post_cls, mask, iou_pred = rid.squeeze(), id.squeeze(), info.squeeze(), \
```

Comment:
return mask_score instread of iou_pred, and rename new_score as mask_score. For mask_rcnn without rescoring, simply return None or post_cls_score. @RogerChern Agree?